### ISSUE TITLE
Asynchronous RF operations

### ISSUE THREAD

**Issue by g-oikonomou** [contiki-ng/contiki-ng#220]
Currently all our RF operations are synchronous.

Especially for TX, we block waiting for potentially a very long time. It would be nice to be able to return immediately at the start of the TX operation, with the radio notifying about the end of TX through a callback.

A similar technique could be applied to the `on()` operation, in the case of radios that have a very long startup time, e.g. due to a need to fire up a crystal.

**Comment by arurke** [comment]
:+1: 
For what is worth, I remember @bkozak-scanimetrics implemented an async radio driver for cc2650 as part of https://github.com/contiki-os/contiki/pull/1549 . I never got around to testing it myself though.

**Comment by yatch** [comment]
Interesting; this would need some adjustments on MAC implementations including TSCH, wouldn't it?

**Comment by g-oikonomou** [comment]
Yep, MAC implementations as well as radio drivers to get those TX notifications.

**Comment by yatch** [comment]
@g-oikonomou OK; thanks! 😄
